### PR TITLE
brave: darwin support, add darwin maintainer

### LINKED ISSUE #352692: Package request: brave for darwin
Brave is already in nixpkgs, but i've spent the past 12 hours trying to get brave to package for darwin aarch64.

I'd pretty much appreciate if someone could package it for darwin as i'm not able enough (apparently).

[here](https://github.com/fbettag/nixpkgs/commit/84760e3e928acc08d28fc06c7f0ab0c864d5cfcc) is my attempt. at least the fetching of checksums worked

### PR DESCRIPTION
**Pull request by matteo-pacini**
Added Darwin support to the Brave Browser derivation, fixes #352692 .

Changes:
- Update script: added Darwin support, and also made sure there's only one source of truth for all platforms (Github Releases)
- Added darwin support to `make-brave.nix`
- Added myself as maintainer for Darwin

Looking forward to suggestions and improvements 🙏🏻 

Notes:
- Needs testing on `x86_64-darwin` - for some reason I can't launch the Intel version via Rosetta (@NixOS/darwin-maintainers )

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by isabelroses — APPROVED** [review]

**Comment by fbettag** [comment]
thanks!

**Comment by siriobalmelli** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352880`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

**Comment by siriobalmelli** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352880`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

**Review by siriobalmelli — APPROVED** [review]

**Comment by matteo-pacini** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352880`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

**Comment by matteo-pacini** [comment]
Just rebased

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352880`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

**Comment by buckley310** [comment]
There's a new Brave update out. If I do an update PR for that update that includes RFC-formatting will that make your life easier or harder?
It will certainly shrink the diff, but might need some manual rebasing.

**Comment by matteo-pacini** [comment]
@buckley310 happy to manually rebase on top of the update if it gets merged first - I could also add the update as part of this PR if needed. That would be a good test for the update script.

**Comment by matteo-pacini** [comment]
@buckley310 rebased on top of your work, ready for re-review.

File: pkgs/by-name/br/brave/make-brave.nix [hunk 1]
```diff
@@ -76,7 +80,6 @@
   version,
   hash,
   url,
-  platform,
```
Comment (matteo-pacini) [inline pkgs/by-name/br/brave/make-brave.nix:79]: Removed this one as I personally think `stdenv.hostPlatform` is more idiomatic

**Comment by matteo-pacini** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352880`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>brave</li>
  </ul>
</details>

---
### `x86_64-linux`

**Pull request merged**